watchos and watchsimulator broken (PR included)

Compiling hello world with `-D watchos` and `-D watchsimulator` generates compilation errors. Adding `-std=c++11` and bumping `-mwatchos-simulator-version-min` to `3.0` resolves all errors. (PR #953)

(hxcpp 4.2.1)

```haxe
function main() {
	trace('hello world');
}
```

**build.hxml**
```hxml
-main Main
--dce full
-cpp bin
```

Full output with `-D HXCPP_VERBOSE`
```
in ~/Downloads/watchos-test ⋮ haxe build.hxml -D HXCPP_VERBOSE
haxelib run hxcpp Build.xml haxe -DHXCPP_VERBOSE="1" -Dhaxe="4.3.0-rc.1" -Dhaxe3="1" -Dhaxe4="1" -Dhaxe_ver="4.300" -Dhxcpp_api_level="400" -Dhxcpp_smart_strings="1" -Dsource-header="Generated by Haxe 4.3.0-rc.1" -Dstatic="1" -Dtarget.name="cpp" -Dtarget.static="true" -Dtarget.sys="true" -Dtarget.threaded="true" -Dtarget.unicode="true" -Dtarget.utf16="true" -Dutf16="1" -Dwatchos="1" -I"src/" -I"" -I"/usr/local/lib/haxe/extraLibs/" -I"/usr/local/share/haxe/extraLibs/" -I"/usr/local/bin/extraLibs/" -I"/usr/local/lib/haxe/std/cpp/_std/" -I"/usr/local/share/haxe/std/cpp/_std/" -I"/usr/local/bin/std/cpp/_std/" -I"/usr/local/lib/haxe/std/" -I"/usr/local/share/haxe/std/" -I"/usr/local/bin/std/"
 __                          
/\ \                                      
\ \ \___    __  _   ___   _____   _____   
 \ \  _ `\ /\ \/'\ /'___\/\ '__`\/\ '__`\ 
  \ \ \ \ \\/>  <//\ \__/\ \ \L\ \ \ \L\ \
   \ \_\ \_\/\_/\_\ \____\\ \ ,__/\ \ ,__/
    \/_/\/_/\//\/_/\/____/ \ \ \/  \ \ \/ 
                            \ \_\   \ \_\ 
                             \/_/    \/_/ 

hxcpp (Haxe C++ Runtime Support) (4.2.1)

Using makefile: Build.xml
Reading HXCPP config: /Users/geo/.hxcpp_config.xml
No specified toolchain

 - Parsing include: /Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/toolchain/setup.xml
 - Parsing include: /Users/geo/.hxcpp_config.xml (section "vars")
 - Running process: xcode-select --print-path
 - Parsing include: /Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/toolchain/finish-setup.xml
 - Parsing makefile: /Users/geo/Downloads/watchos-test/bin/Build.xml
 - Parsing include: /Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/build-tool/BuildCommon.xml
 - Parsing include: /Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/toolchain/haxe-target.xml
 - Parsing include: /Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/toolchain/watchos-toolchain.xml
 - Parsing include: /Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/toolchain/gcc-toolchain.xml
 - Parsing compiler: /Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/toolchain/common-defines.xml
 - Parsing include: /Users/geo/.hxcpp_config.xml (section "exes")

Using compile threads: 4

Ignoring compiler cache because HXCPP_COMPILE_CACHE is not valid.
Creating PCH directory "/Users/geo/Downloads/watchos-test/bin/obj/watchos/__pch/haxe"
 - Precompile /Users/geo/Downloads/watchos-test/bin/obj/watchos/__pch/haxe/hxcpp.h.gch
 - Running command: xcrun --toolchain WatchOS clang++ -Iinclude -c -arch armv7k -stdlib=libc++ -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS6.1.sdk -fapplication-extension -O2 -fmessage-length=0 -pipe -Wno-trigraphs -fno-stack-protector -Wno-overflow -fpascal-strings -fasm-blocks -Wreturn-type -mwatchos-version-min=2.2 -DHX_APPLEWATCH -DHX_WATCHOS -fobjc-arc -DOBJC_ARC -DENABLE_BITCODE=YES -fembed-bitcode -Wno-parentheses -Wno-null-dereference -Wno-unused-value -Wno-bool-conversion -DHXCPP_VISIT_ALLOCS -DHX_SMART_STRINGS -DHXCPP_API_LEVEL=400 -I/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include -fexceptions -fstrict-aliasing -Wno-invalid-offsetof -x c++-header -o /Users/geo/Downloads/watchos-test/bin/obj/watchos/__pch/haxe/hxcpp.h.gch /Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include/hxcpp.h
 - Running command: xcrun --toolchain WatchOS clang++ -I/Users/geo/Downloads/watchos-test/bin/obj/watchos/__pch/haxe -Iinclude -c -arch armv7k -stdlib=libc++ -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS6.1.sdk -fapplication-extension -O2 -fmessage-length=0 -pipe -Wno-trigraphs -fno-stack-protector -Wno-overflow -fpascal-strings -fasm-blocks -Wreturn-type -mwatchos-version-min=2.2 -DHX_APPLEWATCH -DHX_WATCHOS -fobjc-arc -DOBJC_ARC -DENABLE_BITCODE=YES -fembed-bitcode -Wno-parentheses -Wno-null-dereference -Wno-unused-value -Wno-bool-conversion -DHXCPP_VISIT_ALLOCS -DHX_SMART_STRINGS -DHXCPP_API_LEVEL=400 -I/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include -fexceptions -fstrict-aliasing -x c++ -Wno-invalid-offsetof ./src/Std.cpp -o/Users/geo/Downloads/watchos-test/bin/obj/watchos/ceb502b0_Std.o
 - Running command: xcrun --toolchain WatchOS clang++ -I/Users/geo/Downloads/watchos-test/bin/obj/watchos/__pch/haxe -Iinclude -c -arch armv7k -stdlib=libc++ -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS6.1.sdk -fapplication-extension -O2 -fmessage-length=0 -pipe -Wno-trigraphs -fno-stack-protector -Wno-overflow -fpascal-strings -fasm-blocks -Wreturn-type -mwatchos-version-min=2.2 -DHX_APPLEWATCH -DHX_WATCHOS -fobjc-arc -DOBJC_ARC -DENABLE_BITCODE=YES -fembed-bitcode -Wno-parentheses -Wno-null-dereference -Wno-unused-value -Wno-bool-conversion -DHXCPP_VISIT_ALLOCS -DHX_SMART_STRINGS -DHXCPP_API_LEVEL=400 -I/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include -fexceptions -fstrict-aliasing -x c++ -Wno-invalid-offsetof ./src/__boot__.cpp -o/Users/geo/Downloads/watchos-test/bin/obj/watchos/ceb502b0___boot__.o
 - Running command: xcrun --toolchain WatchOS clang++ -I/Users/geo/Downloads/watchos-test/bin/obj/watchos/__pch/haxe -Iinclude -c -arch armv7k -stdlib=libc++ -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS6.1.sdk -fapplication-extension -O2 -fmessage-length=0 -pipe -Wno-trigraphs -fno-stack-protector -Wno-overflow -fpascal-strings -fasm-blocks -Wreturn-type -mwatchos-version-min=2.2 -DHX_APPLEWATCH -DHX_WATCHOS -fobjc-arc -DOBJC_ARC -DENABLE_BITCODE=YES -fembed-bitcode -Wno-parentheses -Wno-null-dereference -Wno-unused-value -Wno-bool-conversion -DHXCPP_VISIT_ALLOCS -DHX_SMART_STRINGS -DHXCPP_API_LEVEL=400 -I/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include -fexceptions -fstrict-aliasing -x c++ -Wno-invalid-offsetof ./src/haxe/Log.cpp -o/Users/geo/Downloads/watchos-test/bin/obj/watchos/e8eae3d8_Log.o
 - Running command: xcrun --toolchain WatchOS clang++ -I/Users/geo/Downloads/watchos-test/bin/obj/watchos/__pch/haxe -Iinclude -c -arch armv7k -stdlib=libc++ -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS6.1.sdk -fapplication-extension -O2 -fmessage-length=0 -pipe -Wno-trigraphs -fno-stack-protector -Wno-overflow -fpascal-strings -fasm-blocks -Wreturn-type -mwatchos-version-min=2.2 -DHX_APPLEWATCH -DHX_WATCHOS -fobjc-arc -DOBJC_ARC -DENABLE_BITCODE=YES -fembed-bitcode -Wno-parentheses -Wno-null-dereference -Wno-unused-value -Wno-bool-conversion -DHXCPP_VISIT_ALLOCS -DHX_SMART_STRINGS -DHXCPP_API_LEVEL=400 -I/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include -fexceptions -fstrict-aliasing -x c++ -Wno-invalid-offsetof ./src/_Main/Main_Fields_.cpp -o/Users/geo/Downloads/watchos-test/bin/obj/watchos/f05ec151_Main_Fields_.o

Error: Error while running command
xcrun --toolchain WatchOS clang++ -I/Users/geo/Downloads/watchos-test/bin/obj/watchos/__pch/haxe -Iinclude -c -arch armv7k -stdlib=libc++ -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS6.1.sdk -fapplication-extension -O2 -fmessage-length=0 -pipe -Wno-trigraphs -fno-stack-protector -Wno-overflow -fpascal-strings -fasm-blocks -Wreturn-type -mwatchos-version-min=2.2 -DHX_APPLEWATCH -DHX_WATCHOS -fobjc-arc -DOBJC_ARC -DENABLE_BITCODE=YES -fembed-bitcode -Wno-parentheses -Wno-null-dereference -Wno-unused-value -Wno-bool-conversion -DHXCPP_VISIT_ALLOCS -DHX_SMART_STRINGS -DHXCPP_API_LEVEL=400 -I/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include -fexceptions -fstrict-aliasing -x c++ -Wno-invalid-offsetof ./src/haxe/Log.cpp -o/Users/geo/Downloads/watchos-test/bin/obj/watchos/e8eae3d8_Log.o

In file included from ./src/haxe/Log.cpp:5:
include/Std.h:46:3: error: found '<::' after a template name which forms the digraph '<:' (aka '[') and a ':', did you mean '< ::'?
                HX_DO_RTTI_ALL;
                ^
/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include/hx/Macros.h:92:26: note: expanded from macro 'HX_DO_RTTI_ALL'
   static ::hx::ObjectPtr<::hx::Class_obj> __mClass; \
                         ^
In file included from ./src/haxe/Log.cpp:5:
include/Std.h:46:3: error: found '<::' after a template name which forms the digraph '<:' (aka '[') and a ':', did you mean '< ::'?
/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include/hx/Macros.h:93:19: note: expanded from macro 'HX_DO_RTTI_ALL'
   ::hx::ObjectPtr<::hx::Class_obj > __GetClass() const { return __mClass; } \
                  ^
In file included from ./src/haxe/Log.cpp:5:
include/Std.h:46:3: error: found '<::' after a template name which forms the digraph '<:' (aka '[') and a ':', did you mean '< ::'?
/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include/hx/Macros.h:94:33: note: expanded from macro 'HX_DO_RTTI_ALL'
   inline static ::hx::ObjectPtr<::hx::Class_obj> &__SGetClass() { return __mClass; } \
                                ^
In file included from ./src/haxe/Log.cpp:8:
include/haxe/Log.h:47:3: error: found '<::' after a template name which forms the digraph '<:' (aka '[') and a ':', did you mean '< ::'?
                HX_DO_RTTI_ALL;
                ^
/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include/hx/Macros.h:92:26: note: expanded from macro 'HX_DO_RTTI_ALL'
   static ::hx::ObjectPtr<::hx::Class_obj> __mClass; \
                         ^
In file included from ./src/haxe/Log.cpp:8:
include/haxe/Log.h:47:3: error: found '<::' after a template name which forms the digraph '<:' (aka '[') and a ':', did you mean '< ::'?
/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include/hx/Macros.h:93:19: note: expanded from macro 'HX_DO_RTTI_ALL'
   ::hx::ObjectPtr<::hx::Class_obj > __GetClass() const { return __mClass; } \
                  ^
In file included from ./src/haxe/Log.cpp:8:
include/haxe/Log.h:47:3: error: found '<::' after a template name which forms the digraph '<:' (aka '[') and a ':', did you mean '< ::'?
/Users/geo/Downloads/watchos-test/.haxelib/hxcpp/4,2,1/include/hx/Macros.h:94:33: note: expanded from macro 'HX_DO_RTTI_ALL'
   inline static ::hx::ObjectPtr<::hx::Class_obj> &__SGetClass() { return __mClass; } \
                                ^
6 errors generated.
Error: Build failed
in ~/Downloads/watchos-test ⋮ 
```

From this thread https://community.haxe.org/t/missing-hxcpp-h-in-cpp-compilation/2975/7

--- a/toolchain/watchos-toolchain.xml
+++ b/toolchain/watchos-toolchain.xml
@@ -14,15 +14,15 @@
   <exe name="xcrun --toolchain WatchOS clang++" />
   <flag value="-c"/>
   <flag value="-arch"/>
-  <flag value="armv7k"/>
-
+  <flag value="armv7k" unless="HXCPP_ARM64_32"/>
+  <flag value="arm64_32" if="HXCPP_ARM64_32"/>
 
   <pchflag value="-x" />
   <pchflag value="c++-header" />
 
   <flag value="-stdlib=libc++" if="HXCPP_CPP11" />
   <flag value="-std=c++11" if="HXCPP_CPP11" />
-  
+
   <flag value="-g" />
 
   <flag value="-isysroot"/>

--- a/toolchain/watchos-toolchain.xml
+++ b/toolchain/watchos-toolchain.xml
@@ -44,7 +44,7 @@
   <mmflag value="-fobjc-abi-version=2"/>
   <mmflag value="-fobjc-legacy-dispatch"/>
 
-  <flag value="-mwatchos-version-min=2.2"/>
+  <flag value="-mwatchos-version-min=3.0"/>
   <flag value="-DHX_APPLEWATCH"/>
   <flag value="-DHX_WATCHOS"/>
   <flag value="-fobjc-arc"  />

--- a/toolchain/watchos-toolchain.xml
+++ b/toolchain/watchos-toolchain.xml
@@ -13,9 +13,14 @@
 <compiler id="WatchOS" exe="clang" >
   <exe name="xcrun --toolchain WatchOS clang++" />
   <flag value="-c"/>
+
   <flag value="-arch"/>
-  <flag value="armv7k" unless="HXCPP_ARM64_32"/>
-  <flag value="arm64_32" if="HXCPP_ARM64_32"/>
+  <flag value="${HXCPP_ARCH}" if="HXCPP_ARCH"/>
+  <section unless="HXCPP_ARCH">
+    <flag value="armv7k" unless="HXCPP_ARM64_32"/>
+    <flag value="arm64_32" if="HXCPP_ARM64_32"/>
+  </section>
+  
 
   <pchflag value="-x" />
   <pchflag value="c++-header" />

--- a/toolchain/watchos-toolchain.xml
+++ b/toolchain/watchos-toolchain.xml
@@ -21,6 +21,8 @@
   <pchflag value="c++-header" />
 
   <flag value="-stdlib=libc++" if="HXCPP_CPP11" />
+  <flag value="-std=c++11" if="HXCPP_CPP11" />
+  
   <flag value="-g" />
 
   <flag value="-isysroot"/>

--- a/toolchain/watchsimulator-toolchain.xml
+++ b/toolchain/watchsimulator-toolchain.xml
@@ -14,15 +14,14 @@
   <exe name="xcrun --toolchain WatchSimulator clang++" />
   <flag value="-c"/>
   <flag value="-arch"/>
-  <flag value="i386"/>
-
+  <flag value="i386" unless="HXCPP_M64"/>
+  <flag value="x86_64" if="HXCPP_M64"/>
 
   <pchflag value="-x" />
   <pchflag value="c++-header" />
 
   <flag value="-stdlib=libc++" if="HXCPP_CPP11" />
   <flag value="-std=c++11" if="HXCPP_CPP11" />
-
   <flag value="-g" />
 
   <flag value="-isysroot"/>
@@ -44,7 +43,7 @@
   <mmflag value="-fobjc-abi-version=2"/>
   <mmflag value="-fobjc-legacy-dispatch"/>
 
-  <flag value="-mwatchos-simulator-version-min=2.2"/>
+  <flag value="-mwatchos-simulator-version-min=3.0"/>
   <flag value="-DHX_APPLEWATCH"/>
   <flag value="-DHX_WATCHSIM"/>
   <flag value="-fobjc-arc"  />

--- a/toolchain/watchsimulator-toolchain.xml
+++ b/toolchain/watchsimulator-toolchain.xml
@@ -17,8 +17,8 @@
   <flag value="-arch"/>
   <flag value="${HXCPP_ARCH}" if="HXCPP_ARCH"/>
   <section unless="HXCPP_ARCH">
-    <flag value="i386" unless="HXCPP_M64"/>
-    <flag value="x86_64" if="HXCPP_M64" unless="HXCPP_ARM64"/>
+    <flag value="i386" unless="HXCPP_M64 || HXCPP_ARM64 || HXCPP_X86_64"/>
+    <flag value="x86_64" if="HXCPP_M64 || HXCPP_X86_64"/>
     <flag value="arm64" if="HXCPP_ARM64"/>
   </section>
 

--- a/toolchain/watchsimulator-toolchain.xml
+++ b/toolchain/watchsimulator-toolchain.xml
@@ -13,9 +13,14 @@
 <compiler id="WatchSimulator" exe="clang" >
   <exe name="xcrun --toolchain WatchSimulator clang++" />
   <flag value="-c"/>
+
   <flag value="-arch"/>
-  <flag value="i386" unless="HXCPP_M64"/>
-  <flag value="x86_64" if="HXCPP_M64"/>
+  <flag value="${HXCPP_ARCH}" if="HXCPP_ARCH"/>
+  <section unless="HXCPP_ARCH">
+    <flag value="i386" unless="HXCPP_M64"/>
+    <flag value="x86_64" if="HXCPP_M64" unless="HXCPP_ARM64"/>
+    <flag value="arm64" if="HXCPP_ARM64"/>
+  </section>
 
   <pchflag value="-x" />
   <pchflag value="c++-header" />

--- a/toolchain/watchsimulator-toolchain.xml
+++ b/toolchain/watchsimulator-toolchain.xml
@@ -21,6 +21,8 @@
   <pchflag value="c++-header" />
 
   <flag value="-stdlib=libc++" if="HXCPP_CPP11" />
+  <flag value="-std=c++11" if="HXCPP_CPP11" />
+
   <flag value="-g" />
 
   <flag value="-isysroot"/>

